Makefile

```diff
@@ -13,160 +13,76 @@
 
 # Check that at most one flowdb service is present in DOCKER_SERVICES
 NUM_SPECIFIED_FLOWDB_SERVICES=$(words $(filter flowdb%, $(DOCKER_SERVICES)))
 ifneq ($(NUM_SPECIFIED_FLOWDB_SERVICES),0)
   ifneq ($(NUM_SPECIFIED_FLOWDB_SERVICES),1)
-    $(error "At most one flowdb service must be specified in DOCKER_SERVICES, but found: $(filter flowdb%, $(DOCKER_SERVICES))")
+	$(error "At most one flowdb service must be specified in DOCKER_SERVICES, but found: $(filter flowdb%, $(DOCKER_SERVICES))")
   endif
 endif
 
+
+ifeq ($(FLOWDB_SERVICE), flowdb_testdata)
+ up up-no_build : DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_TESTDATA_FILE)
+ up : DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_TESTDATA_FILE_BUILD)
+endif
+ifeq ($(FLOWDB_SERVICE), flowdb_synthetic_data)
+ up up-no_build : DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_SYNTHETICDATA_FILE)
+ up : DOCKER_COMPOSE += -f $(DOCKER_COMPOSE_SYNTHETICDATA_FILE_BUILD)
+endif
+
 all:
 
 up: flowdb-build
-	$(DOCKER_COMPOSE_UP) $(DOCKER_SERVICES)
+	$(DOCKER_COMPOSE) up --build -d $(DOCKER_SERVICES_TO_START)
 
 up-no_build:
-	$(DOCKER_COMPOSE_UP_NO_BUILD) $(DOCKER_SERVICES)
+	$(DOCKER_COMPOSE) up -d $(DOCKER_SERVICES_TO_START)
 
 down:
-	$(DOCKER_COMPOSE_DOWN)
-
-
-# Note: the targets below are repetitive and could be simplified by using
-# a pattern rule as follows:
-#
-#   %-up: %-build
-#       docker-compose -f $(DOCKER_COMPOSE_FILE) up -d --build $*
-#
-# The reason we are keeping the explicitly spelled-out versions is in order
-# to increase discoverability of the available Makefile targets and to enable
-# tab-completion of targets (which is not possible when using patterns).
-
-
-flowdb-up: flowdb-build
-	$(DOCKER_COMPOSE_UP) flowdb
-
-flowdb-down:
-	$(DOCKER_COMPOSE_DOWN) flowdb
-
-flowdb-build:
-	$(DOCKER_COMPOSE_BUILD) flowdb
-
-
-flowdb_testdata-up: flowdb_testdata-build
-	$(DOCKER_COMPOSE_UP) flowdb_testdata
-
-flowdb_testdata-down:
-	$(DOCKER_COMPOSE_DOWN) flowdb_testdata
-
-flowdb_testdata-build: flowdb-build
-	$(DOCKER_COMPOSE_BUILD) flowdb_testdata
-
-
-flowdb_synthetic_data-up: flowdb_synthetic_data-build
-	$(DOCKER_COMPOSE_UP) flowdb_synthetic_data
-
-flowdb_synthetic_data-down:
-	$(DOCKER_COMPOSE_DOWN) flowdb_synthetic_data
-
-flowdb_synthetic_data-build: flowdb-build
-	$(DOCKER_COMPOSE_BUILD) flowdb_synthetic_data
-
-
-flowmachine-up:
-	$(DOCKER_COMPOSE_UP) flowmachine
-
-flowmachine-down:
-	$(DOCKER_COMPOSE_DOWN) flowmachine
-
-flowmachine-build:
-	$(DOCKER_COMPOSE_BUILD) flowmachine
-
-
-flowapi-up:
-	$(DOCKER_COMPOSE_UP) flowapi
-
-flowapi-down:
-	$(DOCKER_COMPOSE_DOWN) flowapi
-
-flowapi-build:
-	$(DOCKER_COMPOSE_BUILD) flowapi
-
-
-flowauth-up:
-	$(DOCKER_COMPOSE_UP) flowauth
-
-flowauth-down:
-	$(DOCKER_COMPOSE_DOWN) flowauth
-
-flowauth-build:
-	$(DOCKER_COMPOSE_BUILD) flowauth
-
-
-worked_examples-up: worked_examples-build
-	$(DOCKER_COMPOSE_UP) worked_examples
-
-worked_examples-down:
-	$(DOCKER_COMPOSE_DOWN) worked_examples
+	$(DOCKER_COMPOSE) down
```

Comment:
Previously  `make down` also removed volumes (i.e. `docker-compose down -v`). Is this a deliberate change?